### ISSUE TITLE
How can we set the socket timeout on http(s) request?

### ISSUE THREAD

**Issue by jneuhauser** [imageio/imageio#533]
Hello,

I´m using this great library for fetching random images from unsplash like:

```
import imageio

def unsplash_image_generator():
    while True:
        try:
            image = imageio.imread('https://source.unsplash.com/random')
        except OSError as e:
            print("Error fetching random unsplash image: ", e)
        else:
            yield image
```

Sadly on 6 of 8000 images there is a socket.timeout exception thrown and i wan´t to reduce the timeout to a short time like 3 seconds.
How can this be done and keep the simplicity of this great library?

Thanks in advance.

**Comment by almarklein** [comment]
That timeout is hardcoded here: https://github.com/imageio/imageio/blob/master/imageio/core/request.py#L347

It would be nice to allow specifying the timeout. I think it makes the most sense to use an environment variable for that (https://imageio.readthedocs.io/en/stable/envvariables.html). Care for a PR? ;)

**Comment by jneuhauser** [comment]
I've created a PR that uses the param timeout throught **kwargs.
I think this is more common for such params like timeout, but on the other side this is more a taste of the developer itself.

**Comment by almarklein** [comment]
#closed by #534
